File: alumni-back/src/main/java/ec/edu/ista/springgc1/model/dto/SurveyQuestionsAnswersStatsDTO.java
package ec.edu.ista.springgc1.model.dto;

import java.io.Serializable;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonView;

import ec.edu.ista.springgc1.view.View;
import lombok.Data;

@Data
public class SurveyQuestionsAnswersStatsDTO implements Serializable {
    @JsonView(View.Public.class)
    private Long surveyId;
    @JsonView(View.Public.class)
    private String surveyTitle;
    @JsonView(View.Public.class)
    private String surveyDescription;
    @JsonView(View.Public.class)
    private List<QuestionWithAnswersStatsDTO> questionsWithAnswers;
    @JsonView(View.Public.class)
    private long totalGraduados;
    @JsonView(View.Public.class)
    private int graduadosRespondidos;
    @JsonView(View.Public.class)
    private String careerName;
}

File: alumni_app/src/app/data/model/DTO/SurveyQuestionsAnswersStatsDTO.ts
import { QuestionWithAnswersStatsDTO } from "./QuestionWithAnswersStatsDTO";

export class SurveyQuestionsAnswersStatsDTO{
    surveyId: number;
    surveyTitle: string;
    surveyDescription: string;
    questionsWithAnswers: QuestionWithAnswersStatsDTO[];
    totalGraduados: number;
    graduadosRespondidos: number;
    
    constructor(
      surveyId: number,
      surveyTitle: string,
      surveyDescription: string,
      questionsWithAnswers: QuestionWithAnswersStatsDTO[],
      totalGraduados: number,
      graduadosRespondidos: number
    ) {
      this.surveyId = surveyId;
      this.surveyTitle = surveyTitle;
      this.surveyDescription = surveyDescription;
      this.questionsWithAnswers = questionsWithAnswers;
      this.totalGraduados = totalGraduados;
      this.graduadosRespondidos = graduadosRespondidos;
    }
}